ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MachineAccessIntegrationForm.tsx

```diff
@@ -0,0 +1,377 @@
+    id: '',
+};
+
+function MachineAccessIntegrationForm({
+    initialValues = null,
+    isEditable = false,
+}: IntegrationFormProps<MachineAccessConfig>): ReactElement {
+    const formInitialValues = { ...defaultValues, ...initialValues };
+    const formik = useIntegrationForm<MachineAccessConfig>({
+        initialValues: formInitialValues,
+        validationSchema,
+    });
+    const {
+        values,
+        touched,
+        errors,
+        dirty,
+        isValid,
+        setFieldValue,
+        handleBlur,
+        isSubmitting,
+        isTesting,
+        onSave,
+        onCancel,
+        message,
+    } = formik;
+
+    function onChange(value, event) {
+        return setFieldValue(event.target.id, value);
+    }
+
+    const [roles, setRoles] = useState<Role[]>([]);
+    const [alertRoles, setAlertRoles] = useState<ReactElement | null>(null);
+
+    useEffect(() => {
+        fetchRolesAsArray()
+            .then((rolesFetched) => {
+                setRoles(rolesFetched);
+                setAlertRoles(null);
+            })
+            .catch((error) => {
+                setAlertRoles(
+                    <Alert title="Fetch roles failed" variant={AlertVariant.warning} isInline>
+                        {error.message}
+                    </Alert>
+                );
+            });
+    }, []);
+
+    return (
+        <>
+            {alertRoles}
+            <PageSection variant="light" isFilled hasOverflowScroll>
+                <FormMessage message={message} />
+                <Form isWidthLimited>
+                    <FormikProvider value={formik}>
+                        <FormLabelGroup
+                            isRequired
+                            label="Select configuration type"
+                            fieldId="type"
+                            touched={touched}
+                            errors={errors}
+                        >
+                            <SelectSingle
+                                id="type"
+                                value={values.type}
+                                handleSelect={(name, value) => {
+                                    setFieldValue(name, value);
+                                    setFieldValue(
+                                        'issuer',
+                                        value === 'GITHUB_ACTIONS'
+                                            ? 'https://token.actions.githubusercontent.com'
+                                            : ''
+                                    );
+                                }}
+                                direction="down"
+                                isDisabled={!isEditable}
+                            >
+                                <SelectOption key={'GENERIC'} value={'GENERIC'}>
+                                    Generic
+                                </SelectOption>
+                                <SelectOption key={'GITHUB_ACTIONS'} value={'GITHUB_ACTIONS'}>
```

Comment:
Something the UI team has found helpful lately is to add an `as const` directive to a TypeScript definition for a defined set of allowed values like this, and then when you import that type you can use the TS values as **constant values** in the code, so less repetition and space for typos.

[Examples](https://github.com/stackrox/stackrox/blob/0a7614a0d60478d568cdabd0f0b2e7e507db5124/ui/apps/platform/src/services/AdministrationEventsService.ts#L40-L60), again from the recent new feature AdminstrationEvents.

cc: @pedrottimark can elaborate on this pattern

Reply:
Unfortunately, I didn't understand how adding `as const` directive could help me reuse these constants here :(